marimo/_runtime/runner/cell_runner.py

```diff
@@ -553,7 +553,7 @@ def _get_blamed_cell(
         blamed_cell = None
         try:
             (blamed_cell, *_) = self.graph.get_defining_cells(ref)
-        except KeyError:
```

Comment:
The `ValueError` reads somewhat mysteriously. This should probably be refactored to not use a `try/except`, and check whether the return value of `get_defining_cells` is empty.